### DISCUSSION TITLE
Does (or could) Halide support Texas Instruments C7x target?

### DISCUSSION THREAD

**Discussion by ivangarcia44** [halide/Halide#8461] [Ideas]
Texas Instruments C7x is latest generation DSP from TI. It has a neural network accelerator (B) and a wide SIMD data path.

Halide is very effective in the deep learning area. Some companies could be benefited if this board is supported for Halide.

If it does not support it, what would it take to add Halide support for it? Or is there a reason why it should not be supported?

Thank you,
Ivan

**Comment by alexreinking** [comment]
As far as I'm aware, upstream LLVM doesn't support TI C7x. If it did, support from the community would be on the table. I know TI uses LLVM/Clang as the basis of their C compiler. They would likely need to be convinced to upstream their backend. Unfortunately, there's not really such a thing as "generic" LLVM so a path of generating LLVM code from Halide directly and using TI's `llvm-as` is unlikely to work.

**Comment by abadams** [comment]
For other DSPs we have extended the Halide C backend and used the DSP vendor's C compiler, so that's viable without LLVM, but it's ugly.

**Reply by ivangarcia44** [reply]
Thank you both Alex and Andrew. I will get back to my group with this information. Thanks
